packages/libs/avector/src/lib.rs

```diff
@@ -0,0 +1,551 @@
+//! Concurrent Read Optimized Vector
```

Comment:
What do you mean with "read optimized"?
As I can see from the description below, you have a linked list with `N` elements per bucket. Read-optimized to me means, that both, index operations and iterating are fast, but none of them are if `N` is small (`size_of<T> * N` < cache line). For index operation you need to go through `i / N` pointers to index `i mod N`, for iterating you have to dereference `length / N` pointers. Compared to a vector, where `N` is infinite, this does not sound read-optimized.

Reply:
Yes, the description isn't as accurate; I will replace it. As you outlined, this type won't ever be as fast as `Vec` in read-only operations (this also depends on factors like: is an `Arc` used, a `RwLock`, or a `Mutex`). I mean that `AVec` is optimized for read access instead of write access. (`AVec` never re-allocates, and write has a lock, while read does not.)